### ISSUE TITLE
NetSurf build issues with Duktape 1.5.0

### ISSUE THREAD

**Issue by svaarala** [svaarala/duktape#764]
Inline issue:

- http://ci.netsurf-browser.org/jenkins/job/netsurf/3536/CC=gcc,HOST=amd64-unknown-openbsd5.7,TARGET=monkey/parsed_console/
- Fixed by http://git.netsurf-browser.org/netsurf.git/commit/?id=510203b246edc70656c404949a86accd54880610
- That fix caused issues with some other compilers if I recall correctly, however

```
duk_util.h:519: warning: 'duk_raw_read_u16_be' declared inline after being called
duk_util.h:519: warning: previous declaration of 'duk_raw_read_u16_be' was here
duk_util.h:520: warning: 'duk_raw_read_u32_be' declared inline after being called
duk_util.h:520: warning: previous declaration of 'duk_raw_read_u32_be' was here
duk_util.h:521: warning: 'duk_raw_read_double_be' declared inline after being called
duk_util.h:521: warning: previous declaration of 'duk_raw_read_double_be' was here
duk_util.h:522: warning: 'duk_raw_write_u16_be' declared inline after being called
duk_util.h:522: warning: previous declaration of 'duk_raw_write_u16_be' was here
duk_util.h:523: warning: 'duk_raw_write_u32_be' declared inline after being called
duk_util.h:523: warning: previous declaration of 'duk_raw_write_u32_be' was here
duk_util.h:524: warning: 'duk_raw_write_double_be' declared inline after being called
duk_util.h:524: warning: previous declaration of 'duk_raw_write_double_be' was here
```

Clang build issue:

- http://ci.netsurf-browser.org/jenkins/job/netsurf/3537/
- http://ci.netsurf-browser.org/jenkins/job/netsurf/3537/CC=clang,HOST=x86_64-linux-gnu,TARGET=monkey/parsed_console/

```
 duk_error_macros.c:61:19: error: unused function 'duk_err_unsupported_defmsg' [-Werror,-Wunused-function]
```

This one is fixed in master by removing `duk_strings.c`, #760.

**Comment by svaarala** [comment]
The `duk_err_unsupported_defmsg` is actually a function (not a string). It's called by error macros when a call site throws an "unsupported" error. The presence of such call sites depends on config options so that it's rather awkward to either include the function in the build or leave it out: the #ifdef for the function needs to track the call site conditions which is quite unmodular. But since this is an isolated case, it's maybe worth fixing. There are a few existing fixes already place, e.g. in debugger code.

**Comment by svaarala** [comment]
This fixes NetSurf (harmless) warning issues: http://source.netsurf-browser.org/netsurf.git/commit/?id=2cbb337756d9af5bda4d594964d446439f602551

**Comment by svaarala** [comment]
Fixed in 1.5.x maintenance branch. Master will have a different fix.
